pkg/service/handler.go

```diff
@@ -92,10 +102,47 @@ func (h *Handler) Run() error {
 			}
 
 			h.sendResponse(ctx, request, p.Info, err)
+		case debugResponseChan := <-h.debugRequests:
```

Comment:
can this happen inside Handler.GetPipelineDebugInfo instead?

Reply:
Yes, if we store a reference to the pipeline inside the Handler object, protected by a mutex, with a null check in GetPipelineDebugInfo. Both options have pros and cons I think and I have no strong opinion. What do you prefer?

Reply:
More specifically: the Mutex option is less code, and less indirections. The channel option may be a more Go like pattern, and I. think there is some readability gained by keeping all accesses to an object within the same function/runloop.